src/components/ImagePreviewer.js

```diff
@@ -0,0 +1,176 @@
+import { stringify } from "querystring";
+import { decode } from "base58";
+
+const noop = () => {};
+
+export const of = src => Promise.resolve({ src });
+
+export const makeSrcResolver = ({ src }) =>
+  resolvers.reduceRight((acc, resolver) => {
+    if (!acc && resolver.test(src)) {
+      return resolver.request(src);
+    }
+    return acc;
+  }, false);
+
+export const makeImageDOMResolver = ({ src }) =>
+  new Promise((resolve, reject) => {
+    const img = new Image();
+    img.onload = () =>
+      resolve({
+        src,
+        height: img.height
+      });
+    img.onerror = reject;
+    img.src = src;
```

Comment:
Is this to preload image or to make sure it's a image? I think it should also do a simple check like in `prePicRel` before sending a request. We don't want to load every link user hovers.

Reply:
Just `preload image`.
The non-parsable links are now handled in promise reject case.

Reply:
I realized that `prePicRel` is no longer called in any places so I removed it (correct me if I'm wrong).

Reply:
I see. Call to that function was removed in 861847203bb2ad82f02a9b1e8aef113b46d71acf, so it currently only loads whitelisted sites. It might be good to resurrect that back later. It doesn't have to be this PR.

Reply:
I see. It's used here:

```js
s0 += '<a scol="'+col+'" class="y q'+this.deffg+' b'+this.defbg+'" href="' +ch.getFullURL() + '"' + this.prePicRel( ch.getFullURL()) + ' rel="noreferrer" target="_blank" srow="'+row+'">';
```

But I can't really tell why we need `prePicRel` with `type=?` on the anchor tag. Mind to open an issue with more detailed description instead?

Reply:
To my understanding, it's used to annotate that the link is an image. Later, the it uses css selector to find all the images and does further processing:
https://github.com/robertabcd/PttChrome/blob/e317cae905f1beffca673180a6e6e6001994f831/src/js/term_view.js#L788

Reply:
Yep. Created #25